src/Commands/TankDrive.cpp

```diff
@@ -42,6 +42,32 @@ void TankDrive::Execute() {
 	RobotMap::chassisrightMotor2->Set(rY);
 	RobotMap::chassisrightMotor3->Set(rY);
 
+	//Automatic Shifters
+	SmartDashboard::PutNumber("EncodeValue chassisrightmotor2",
+			RobotMap::chassisrightMotor2->GetEncVel());
+	double shiftUpVelocity = SmartDashboard::GetNumber("UpVelocity", 3000);
+
+	double shiftDownVelocity = SmartDashboard::GetNumber("DownVelocity", 1000);
+
+	const double kThrottle = 0.8;
+
+	if ((RobotMap::chassisshift->Get() == DoubleSolenoid::kReverse)
+			&& (RobotMap::chassisleftMotor1->GetEncVel() >= shiftUpVelocity
+			|| RobotMap::chassisrightMotor2->GetEncVel() >= shiftUpVelocity)
+		 && (Robot::oi->getdriver()->GetRawAxis(1) >= kThrottle
+				|| Robot::oi->getdriver()->GetRawAxis(5) >= kThrottle) )
+	{
+		RobotMap::chassisshift->Set(DoubleSolenoid::kForward);
+	}
+
+	if ((RobotMap::chassisshift->Get() == DoubleSolenoid::kForward)
```

Comment:
Shouldn't an else if go here? I guess it wouldn't matter much for what is trying to be accomplished.